lib/cog/chat/slack/connector.ex

```diff
@@ -300,12 +300,21 @@ defmodule Cog.Chat.Slack.Connector do
   defp classify_id(id, slack),
     do: id |> String.replace(~r/(^<|>$)/, "") |> do_classify_id(slack)
 
-  # TODO: use Slack.Lookups to verify the "C" and "U" cases here, to
-  # guard against bare user / room names that start with those letters
-  defp do_classify_id(<<"#C", id::binary>>, _), do: {:channel, "C#{id}"}
-  defp do_classify_id(<<"C", id::binary>>, _), do: {:channel, "C#{id}"}
-  defp do_classify_id(<<"@U", id::binary>>, _), do: {:user, "U#{id}"}
-  defp do_classify_id(<<"U", id::binary>>, _), do: {:user, "U#{id}"}
+  # We could get an actual id, U1234567, or the user, @username, or the
+  # channel, #channelname, reference. So we check for everything and
+  # return a normalized id.
+  defp do_classify_id(<<"#C", id::binary>>, slack),
+    do: normalize_channel_id(:by_id, "C#{id}", slack)
+  defp do_classify_id(<<"C", id::binary>>, slack),
```

Comment:
I didn't know about the "G". I just fixed the functionality that was already here, and we weren't dealing with that before. I'll confirm that everything works as expected, but I believe we are already doing a test that check the private group behavior.